src/components/MovieList.vue

```diff
@@ -1,35 +1,29 @@
 <script setup lang="ts">
-import { defineProps, getCurrentInstance } from "vue";
-import { lazyloadDirective } from '../utils.js';
+import { computed, getCurrentInstance } from "vue";
+import { lazyloadDirective } from '../utils.ts';
+import useMovieStore from "../store/movieStore";
+
+const movieStore = useMovieStore();
+const movies = computed(() => movieStore.list);
 
 const instance = getCurrentInstance();
 if(instance){
   const app = instance.appContext.app;
-  app?.directive('lazyload', lazyloadDirective);
+  if(!app.directive('lazyload'))app?.directive('lazyload', lazyloadDirective);
 }
 
-interface Movie {
-  name: string;
-  genre: string;
-  year: number;
-  rating: number;
-  movieLength: number;
-  description: string;
-  poster: string;
-}
-defineProps<{ movies: Movie[] }>();
 </script>
 
 <template>
   <div class="movies">
-    <div v-if="movies.length > 0" class="movies__list">
-      <div v-for="(m, i) in movies" class="card" :key="i">
-        <img class="card__img" v-lazyload="m.poster" :alt="m.name"/>
+    <div v-if="movies" class="movies__list">
+      <div v-for="(m) in movies" class="card" :key="m.id" @click="movieStore.toggleMovieDetailPanel(m.id)">
```

Comment:
Don't include the storage callback here, it's bad approach.
Better to have internal callback in `script`